### PR TITLE
chore(linter): add allowReject option to `unicorn/no-useless-promise-resolve-reject`

### PR DESCRIPTION
**Pull request by no-yan**
followup to #7232.

I'll keep this draft until #7232 is merged.

### CODE REVIEW

**Review by camc314 — COMMENTED** [review]
lgtm

File: crates/oxc_linter/src/rules/unicorn/no_useless_promise_resolve_reject.rs [hunk 1]
```diff
@@ -55,6 +60,17 @@ declare_oxc_lint!(
 );
 
 impl Rule for NoUselessPromiseResolveReject {
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let config = value.get(0);
+
+        let allow_reject = config
+            .and_then(|c| c.get("allowReject"))
+            .and_then(serde_json::Value::as_bool)
+            .map_or(false, |v| v);
```
Comment (camc314) [inline crates/oxc_linter/src/rules/unicorn/no_useless_promise_resolve_reject.rs:69]: unneeded closure - use `unwrap_or_default` or sometihg like that?

**Comment by no-yan** [comment]
@camc314 Thank you. I’ve made the revisions; could you please take another look?

**Review by camc314 — APPROVED** [review]
thank you!

**Pull request merged**